testsuite/python/widom_insertion.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (C) 2013-2018 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Testmodule for the Widom Insertion.
+"""
+import os
+import sys
+import unittest as ut
+import numpy as np
+import espressomd  # pylint: disable=import-error
+from espressomd import reaction_ensemble
+
+
+class ReactionEnsembleTest(ut.TestCase):
+
+    """Test the implementation of the widom insertion (part of the reaction ensemble)."""
+
+    # The excess chemical potential is calculated for identical particles into
+    # a 20 cubed box with a single particle, interacting via a LJ-potential
+    # (cut-off at 5 sigma).
+    N0 = 1
+    temperature = 0.5
+    type_HA = 0
+    charge_HA = 0
+    lj_eps = 1.0
+    lj_sig = 1.0
+    lj_cut = 5.0
+    box_l = 20.0
+
+    target_mu_ex = -0.00227888  # analytical solution for this specific system
+
+    system = espressomd.System(box_l=np.ones(3) * box_l)
+    system.seed = system.cell_system.get_state()['n_nodes'] * [2]
+    np.random.seed(69)  # make reaction code fully deterministic
+    system.cell_system.skin = 0.4
+    volume = np.prod(system.box_l)  # cuboid box
+    
+    unimportant_exclusion_radius = 1.0
```

Comment:
Yes widom insertion does not make use of the exclusion radius. So lets just put np.nan as argument for the exclusion radius (I plan to remove this argument from the widom insertion constructor).